### PR TITLE
tags

### PR DESCRIPTION
**Pull request by bpasfinsight**

### CODE REVIEW

File: src/Annotation/Tags.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace Dealroadshow\K8S\Framework\Annotation;
```
Comment (petr-buchyn) [inline src/Annotation/Tags.php:3]: Annotation is name, used earlier.
PHP feature is called Attributes, so I think we should name this namespace `Attribute`

File: src/Annotation/Tags.php [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+    public function __construct(string ...$tags)
+    {
+        $this->tags = $tags;
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getTags(): array
```
Comment (petr-buchyn) [inline src/Annotation/Tags.php:23]: I think that class is already named Tags, so we can name this method `all()` or `get()`

File: src/Annotation/TagsScanner.php [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    {
+        $reflectionClass = new ReflectionClass($object::class);
+        $attributes = $reflectionClass->getAttributes(Tags::class);
+        $tags = [];
+        foreach($attributes as $attribute) {
+            /** @var Tags $tagsAttribute */
+            $tagsAttribute = $attribute->newInstance();
+            $tags = array_merge($tags, $tagsAttribute->getTags());
+        }
+        $tags = array_unique($tags);
```
Comment (petr-buchyn) [inline src/Annotation/TagsScanner.php:25]: empty line before `return`

File: src/Registry/Query/ManifestsQuery.php [hunk 4]
```diff
@@ -34,6 +35,26 @@ public function app(AppInterface $app): self
         return $this->namespacePrefix($reflection->getNamespaceName());
     }
 
+    public function includeTags(array $tags): self
+    {
+        return $this->addClosure(
+            function (ManifestInterface $manifest) use($tags): bool {
```
Comment (petr-buchyn) [inline src/Registry/Query/ManifestsQuery.php:41]: space after `use`

File: src/Registry/Query/ManifestsQuery.php [hunk 5]
```diff
@@ -34,6 +35,26 @@ public function app(AppInterface $app): self
...
+                $manifestTags = TagsScanner::tags($manifest);
+                return count(array_intersect($tags, $manifestTags)) > 0;
+            }
+        );
+    }
+
+    public function excludeTags(array $tags): self
+    {
+        return $this->addClosure(
+            function (ManifestInterface $manifest) use($tags): bool {
```
Comment (petr-buchyn) [inline src/Registry/Query/ManifestsQuery.php:51]: space after `use`

File: src/Attribute/TagsScanner.php [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+use ReflectionClass;
+
+class TagsScanner
+{
+    /**
+     * @param object $object
+     * @return string[]
+     * @throws \ReflectionException
+     */
+    public static function tags(object $object): array
```
Comment (petr-buchyn) [inline src/Attribute/TagsScanner.php:15]: I think this better be named `scan()`, since class is named `TagsScanner`

**Review by petr-buchin — APPROVED** [review]

**Review by petr-buchyn — APPROVED** [review]

**Pull request merged**